src/history_function.jl

```diff
@@ -24,8 +24,23 @@ function (f::HistoryFunction)(t, deriv::Type=Val{0}, idxs=nothing)
         end
     elseif t <= f.sol.t[end] # Put equals back
         return f.sol.interp(t, idxs, deriv)
+    end
+
+    integrator = f.integrator
+
+    # set boolean that indicates that history function was evaluated at time point past the
+    # final time point of the current solution
+    integrator.isout = true
```

Comment:
This is usually for `isoutofdomain`. It is confusing to repurpose it like this. Add another boolean to the integrator instead since that's really cheap anyways.

Reply:
I know, but it's an object of type `ODEIntegrator` that is never iterated and is only used for inter/extrapolation - so we would have to change it in OrdinaryDiffEq. I thought it's not worth it for just this special use case in DelayDiffEq. Shall I add a field to `ODEIntegrator` nevertheless?

Reply:
I guess it should be fine. It doesn't interfere with `isoutofdomain` though?